src/serializer/ResidualFunctions.js

```diff
@@ -276,13 +284,14 @@ export class ResidualFunctions {
           }
           funcNode.body.body = scopeInitialization.concat(funcNode.body.body);
 
-          traverse(t.file(t.program([funcNode])), ClosureRefReplacer, null, {
-            serializedBindings,
-            modified,
-            requireReturns: this.requireReturns,
-            requireStatistics,
-            isRequire: this.modules.getIsRequire(funcParams, [functionValue]),
-          });
+          if (!rewrittenBody)
+            traverse(t.file(t.program([funcNode])), ClosureRefReplacer, null, {
```

Comment:
I'll put these github comments into code when I have a more final version. Here we're also trying to avoid the name rewriting because the body no longer has names that correspond to the original program.